dbt/adapters/snowflake/impl.py

```diff
@@ -199,16 +219,17 @@ def submit_python_job(self, parsed_model: dict, compiled_code: str):
 AS
 $$
 {compiled_code}
-
 $$;
-        """
-        self.execute(python_stored_procedure, auto_begin=False, fetch=False)
-        response, _ = self.execute(f"CALL {proc_name}()", auto_begin=False, fetch=False)
-        self.execute(
-            f"drop procedure if exists {proc_name}(string)",
-            auto_begin=False,
-            fetch=False,
-        )
+CALL {proc_name}();
+
+            """
+        response, _ = self.execute(python_stored_procedure, auto_begin=False, fetch=False)
+        if not use_anonymous_sproc:
+            self.execute(
+                f"drop procedure if exists {proc_name}(string)",
```

Comment:
```suggestion
                f"drop procedure if exists {proc_name}()",
```
Jeremiah H (from Snowflake) noticed that this fails to drop the sproc, because it's created without any arguments `()` and then `drop if exists` with one argument `(string)`